next-tavla/src/Admin/scenarios/BoardList/components/BoardListOptions/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { IconButton, SecondarySquareButton } from '@entur/button'
+import { CloseIcon, SettingsIcon } from '@entur/icons'
+import classes from './styles.module.css'
+import {
+    Popover,
+    PopoverCloseButton,
+    PopoverContent,
+    PopoverTrigger,
+} from '@entur/tooltip'
+import { Heading4 } from '@entur/typography'
+import { Checkbox } from '@entur/form'
+import { Dispatch, SetStateAction } from 'react'
+import { uniqueArrayItems } from 'utils/uniqueArrayItems'
+import { TOptionalColumn } from 'types/optionalColumns'
+
+function BoardListOptions({
+    optionalColumns,
+    shownOptionalColumns,
+    setShownOptionalColumns,
+}: {
+    optionalColumns: TOptionalColumn[]
+    shownOptionalColumns: TOptionalColumn[]
+    setShownOptionalColumns: Dispatch<SetStateAction<TOptionalColumn[]>>
+}) {
+    const handleCheckboxChange = (
+        e: React.ChangeEvent<HTMLInputElement>,
+        column: TOptionalColumn,
+    ) => {
+        const checked = e.target.checked
+        if (checked) {
+            setShownOptionalColumns((shownColumns) =>
+                uniqueArrayItems([...shownColumns, column]),
+            )
+        } else {
+            setShownOptionalColumns((shownColumns) =>
+                shownColumns.filter((shownColumn) => shownColumn !== column),
```

Comment:
Syntes man kan lage en `toggle` funksjon for dette, så slipper man "kompleks" logikk her (ref samtalen om reducer)

Reply:
Tenker du da en toggle-function definert i `BoardList` som vi sender ned med props?